### PR TITLE
RISC-V Emulated Testing

### PR DESCRIPTION
**Pull request by tblah**
Adds a disabled docker image on which to run RISC-V tests. Based on the armhf image.

Test using
```
./src/ci/docker/run.sh riscv64gc-linux
```

cc: @msizanoen1

### CODE REVIEW

File: src/tools/remote-test-client/src/main.rs [hunk 1]
```diff
@@ -170,19 +212,30 @@ fn start_qemu_emulator(target: &str, rootfs: &Path, server: &Path, tmpdir: &Path
...
+                .arg("user,id=net0,hostfwd=tcp::12345-:12345")
+                .arg("-device")
+                .arg("virtio-net-device,netdev=net0,mac=00:00:00:00:00:00")
+                .arg("-device")
+                .arg("virtio-blk-device,drive=hd0")
+                .arg("-drive")
+                .arg(&format!("file={},format=raw,id=hd0", &rootfs_img.to_string_lossy()));
+            t!(cmd.spawn());
         }
+        _ => panic!("cannot start emulator for: {}" < target),
```
Comment (jfrimmel) [inline src/tools/remote-test-client/src/main.rs:238]:
```suggestion
        _ => panic!("cannot start emulator for: {}", target),
```
This seems to be an error, that was present also in the previous code.

Comment (tblah) [reply src/tools/remote-test-client/src/main.rs]: Thanks

**Comment by pietroalbini** [comment]
This looks fine from a cursory glance. Thanks for sending the PR!

@bors r+

File: src/ci/docker/disabled/riscv64gc-linux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+# Compile the kernel that we're going to be emulating with. This is
+# basically just done to be compatible with the QEMU target that we're going
+# to be using when running tests.
+RUN curl https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.6.16.tar.xz | tar xJf - && \
+    cp linux.config linux-5.6.16/.config && \
+    cd /build/linux-5.6.16 && \
+    make olddefconfig && \
+    make -j$(nproc) vmlinux
+RUN cp linux-5.6.16/vmlinux /tmp
+RUN rm -rf linux-5.6.16
```
Comment (bjorn3) [inline src/ci/docker/disabled/riscv64gc-linux/Dockerfile:45]: If you use `... && make -j$(nproc) vmlinux && cp linux-5.6.16/vmlinux /tmp && rm -rf linux-5.6.16`, the `linux-5.6.16` dir will not become part of the dockee image. Every RUN command creates a new overlay on top of the image as of before the RUN command. It doesn't replace it.

Comment (bjorn3) [reply src/ci/docker/disabled/riscv64gc-linux/Dockerfile:45]: Gentle ping @tblah

Comment (tblah) [reply src/ci/docker/disabled/riscv64gc-linux/Dockerfile:45]: @bjorn3 thanks for the reminder, I'll take a look

**Pull request merged**